src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,201 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
+     */
+    private boolean calculateNextItinerary() {
+        // if the trip is not active on the current zoned date time, advance until a day is found when the trip is
+        // active. It is guaranteed that the trip will be active on a certain date because a call to
+        // {@link MonitoredTrip#isInactive} should already have been made in the shouldSkipMonitoredTripCheck method.
+        while (!trip.isActiveOnDate(targetZonedDateTime)) {
+            targetZonedDateTime = targetZonedDateTime.plusDays(1);
+        }
+        targetDate = targetZonedDateTime.format(DATE_FORMATTER);
+
+        // execute trip plan request for the target time
+        if (!calculateOtpResponse()) {
+            // failed to calculate, return false
+            return false;
+        }
+
+        // check if the resulting itinerary has a matching itinerary
+        if (!findMatchingItinerary()) {
+            // TODO: figure out how to notify the user that the itinerary is not possible anymore
+            // for now, just say the next itinerary couldn't be found
+            return false;
+        }
+        return true;
+    }
+
+    private boolean calculateOtpResponse() {
+        OtpDispatcherResponse otpDispatcherResponse;
+        try {
+            otpDispatcherResponse = OtpDispatcher.sendOtpPlanRequest(generateTripPlanQueryParams());
+        } catch (Exception e) {
+            // TODO: report bugsnag
+            LOG.error("Encountered an error while making a request ot the OTP server. error={}", e);
+            return false;
+        }
+
+        if (otpDispatcherResponse == null) return false;
+
+        if (otpDispatcherResponse.statusCode >= 400) {
+            // TODO: report bugsnag
+            LOG.error("Received an error from the OTP server. status={}", otpDispatcherResponse.statusCode);
+            return false;
+        }
+        otpResponse = otpDispatcherResponse.getResponse();
         return true;
     }
 
     /**
-     * Generate the appropriate OTP query params for the trip for the current check. This currently involves replacing
+     * Generate the appropriate OTP query params for the trip for the current check by replacing
      * the date query parameter with the appropriate date.
-     *
-     * TODO: finish implementation
      */
-    public static String generateTripPlanQueryParams(MonitoredTrip trip) throws URISyntaxException {
+    public String generateTripPlanQueryParams() throws URISyntaxException {
         // parse query params
         List<NameValuePair> params = URLEncodedUtils.parse(
             new URI(String.format("http://example.com/%s", trip.queryParams)),
             UTF_8
         );
 
-        // begin building a new list and along the way and figure out if this trip is a depart at or arrive by trip
+        // building a new list by copying all values, except for the date which is set to the target date
         List<NameValuePair> newParams = new ArrayList<>();
```

Comment:
Echoing my comment for `MonitoredTrip#getParsedParams`, a map might be useful here too.

Reply:
I disagree because this variable is sent to the `URLEncodedUtils.format` which needs some kind of Iterable.

Reply:
This is fine for this PR. I ended up needing to do the map in #77 and using it a dozen times.